### ISSUE TITLE
[Crash] invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by sungwoncho** [dgraph-io/dgraph#1022]
```
Dgraph version   : v0.7.7
Commit SHA-1     : 50664a6
Commit timestamp : 2017-05-25 16:42:31 +1000
Branch           : HEAD
```

---

*Reproduction*

1. Set up the following schema

```
mutation {
  schema {
    AboutMe: string @index(fulltext) .
    Author: uid @reverse .
    DisplayName: string @index(exact) .
    Location: string @index(term) .
    Reputation: int @index(int) .
    Score: int @index .
    Text: string @index(term) .
    Type: string @index(exact) .
    ViewCount: int @index .
    Vote: uid @reverse .
    Title: uid @reverse .
    Body: uid @reverse .
    Post: uid @reverse .
    PostCount: int @index .
    Tag: uid @reverse .
    Timestamp: datetime .
    GitHubID: string @index(exact) .
    Has.Answer: uid @reverse .
    Comment: uid @reverse .
    Upvote: uid @reverse .
    Downvote: uid @reverse .
  }
}
```

2. Load the following data

[dgraph-1-2017-06-07-10-18.rdf.gz](https://github.com/dgraph-io/dgraph/files/1056596/dgraph-1-2017-06-07-10-18.rdf.gz)
[dgraph-schema-1-2017-06-07-10-18.rdf.gz](https://github.com/dgraph-io/dgraph/files/1056597/dgraph-schema-1-2017-06-07-10-18.rdf.gz)

3. Run the following query:

```

{
  
questions(func: eq(Type, "Question"), orderdesc: Timestamp, first: 100) {
  
_uid_

Title {
  Text
}

Owner {
  DisplayName
  Reputation
  _uid_
}

Tags {
  TagName: Text
}

Has.Answer(orderdesc: Timestamp, first: 1) {
  Owner {
    DisplayName
    Reputation
    _uid_
  }
  Timestamp
}

ChosenAnswerCount: count(Chosen.Answer)
UpvoteCount: count(Upvote)
DownvoteCount: count(Downvote)
AnswerCount: count(Has.Answer)
ViewCount
Timestamp

}

  
t as var(func: eq(Type, "Tag")) {
  c as count(~Tags)
}

topTags(id: var(t), orderdesc: var(c), first: 10) {
  _uid_
  TagName: Text
  QuestionCount: var(c)
}

  
var(func: eq(Type, "Question"), orderdesc: Timestamp, first: 100) {
  ca as Chosen.Answer
}

var(id: var(ca)) @groupby(Owner) {
  a as count(_uid_)
}

topUsers(id: var(a), orderdesc: var(a)) {
  _uid_
  AboutMe
  DisplayName
  Reputation
  NumAcceptedAnswers: var(a)
}
}
```

4. Dgraph responds with `Predicate Tags doesn't have reverse edge` and crashes:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x46e3a18]

goroutine 68837 [running]:
github.com/dgraph-io/dgraph/query.(*SubGraph).applyOrderAndPagination(0xc463530480, 0x500b900, 0xc423f28480, 0xc449629720, 0xc4496296e0)
	/Users/travis/gopath/src/github.com/dgraph-io/dgraph/query/query.go:1903 +0x1b8
github.com/dgraph-io/dgraph/query.ProcessGraph(0x500b900, 0xc423f28480, 0xc463530480, 0x0, 0xc423f284e0)
	/Users/travis/gopath/src/github.com/dgraph-io/dgraph/query/query.go:1732 +0x1708
created by github.com/dgraph-io/dgraph/query.ProcessQuery
	/Users/travis/gopath/src/github.com/dgraph-io/dgraph/query/query.go:1234 +0x717
```

**Comment by manishrjain** [comment]
Thanks for the bug report, @sungwoncho. Keep them coming!

**Comment by sungwoncho** [comment]
Was fixed by #993. No longer happens after upgrading to the following:

```
Dgraph version   : v0.7.7
Commit SHA-1     : e5445ee
Commit timestamp : 2017-06-06 12:23:45 +1000
Branch           : HEAD
```
